### PR TITLE
Use Decorator Design Pattern

### PR DESCRIPTION
**Pull request by Elerqsousy**
In this milestone:
- Added Nameable class
- Modified Person class to inherit from Nameable class and added correct_name method to override parent method.
- Added Decorator class that inherits from Nameable class
- Added Two decorator classes inheriting from the Decorator class each returning a modified name string according to their purpose.

### CODE REVIEW

File: nameable.rb [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+class CapitalizeDecorator < Decorator
+  def correct_name
+    @nameable.correct_name.split.map(&:capitalize).join(' ')
+  end
+end
+
+# Class TrimmerDecorator inhereting from Decorator
+class TrimmerDecorator < Decorator
+  def correct_name
+    @nameable.correct_name.strip
```
Comment (zieeco) [inline nameable.rb]: - Hey, using the strip method here won't check the string to ensure the character is not more than 10. You can read more about strip [here](https://www.educative.io/answers/what-is-the-strip-method-in-ruby), instead, you can use another method. Here's an example:

` @nameable.correct_name.split.first(10).join(' ')` or 
`@nameable.correct_name.length > 10 ? @nameable.correct_name[0..9] : @nameable.correct_name`

**Review by zieeco — CHANGES_REQUESTED** [review]
Hi @Elerqsousy ,


Great job so far! 👏 
You've done well, however, there are some issues that you still need to work on to go to the next project, but you are almost there!


## To highlight 🎯

- No linter errors.✔️
- Nice Decoration. ✔️
- Descriptive PR. ✔️


## Required Changes ♻️
_Check the comments under the review._





## Optional suggestions
- N/A

**_Every comment with the **[OPTIONAL]** prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._**


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: nameable.rb [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+
+  def correct_name
+    @nameable.correct_name
+  end
+end
+
+# Class CapitalizeDecorator inhereting from Decorator
+class CapitalizeDecorator < Decorator
+  def correct_name
+    @nameable.correct_name.split.map(&:capitalize).join(' ')
```
Comment (DammyShittu) [inline nameable.rb]: - **[OPTIONAL]** This long method of chaining is somewhat unnecessary. Consider that we can implement this using the following code:

  ```
  @nameable.correct_name.capitalize
  ```

**Review by DammyShittu — APPROVED** [review]
Hi @Elerqsousy,

## APPROVED 🟢

## Highlights
✔️ Descriptive PR
✔️ Passing linters
✔️ Good branch name

Great implementation of all the requirements in this milestone. 👏 

_You have done an amazing job 👍_

Your project is complete!  There is nothing else to say other than... it's time to merge it. :shipit: 

Congratulations! 🎉 

## CHEERS AND HAPPY CODING!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**